docs/introduction/visual-inspector-and-dev-tools.md

```diff
@@ -0,0 +1,245 @@
+
+![Stored](https://cloud.githubusercontent.com/assets/674727/24733850/03f8a96e-1a2f-11e7-8420-34f63c4415a5.png)
+
+Another use case for motion capture is as an animation tool to create paths and
+splines, but we won't go over that here.
+
+## How to Record
+
+Read the [Motion Capture
+documentation](https://github.com/dmarcos/aframe-motion-capture-components/)
+for more information. Here's how to set up the recording:
+
+1. Drop the Motion Capture Components script tag into your HTML file (e.g., `<script src="https://unpkg.com/aframe-motion-capture-components/dist/aframe-motion-capture-components.min.js">`)
+2. Add the `avatar-recorder` component to the scene (i.e., `<a-scene avatar-recorder>`)
+3. Enter VR
+4. Press `<space>` to start recording
+5. Record movements and actions
+6. Press `<space>` to stop recording
+7. Save the recording JSON file or upload it by hitting `u` to get a short URL to share between computers
+
+Now we can replay the recording. You can try recording the camera with WASD and
+mouse drag controls right now on a desktop. Head to the [Record
+Example](http://dmarcos.github.io/aframe-motion-capture-components/), open the
+browser Console to get feedback, and hit `<space>` to start and stop recording!
+
+
+![Record](https://cloud.githubusercontent.com/assets/674727/24766726/3d660b7e-1ab1-11e7-9f64-d97b5732ec43.gif)
+
+## How to Replay
+
+By default, the recording will also be saved and replayed from localStorage. If
+we want to take our recording on the go, here's how to replay a recording
+(assuming we already have the script tag above):
+
+1. Put the recording file somewhere accessible to the web page (i.e., in the project directory or online)
+2. Add `avatar-replayer` component to the scene (i.e., `<a-scene avatar-replayer>`)
+3. Append `?avatar-recording=path/to/recording.json` to the URL *or* set `<a-scene avatar-replayer="src: path/to/recording.json>`
+
+Then replay the recording on any device from anywhere without a headset to your
+heart's content. Get in the headset, record some clicks, and then from a
+laptop, we can build event handlers on top of the controller events we emitted
+in the recording.
+
+## Spectator Mode
+
+The A-Frame Motion Capture Components have a Spectator Mode feature that can be
+enabled from the `avatar-replayer` component:
+
+```
+<a-scene avatar-replayer="spectatorMode: true">
+```
+
+This lets you view the recording from a third-person view. I found this
+immensely useful because it was hard to see what was happening in first-person.
+The first-person view is naturally shaky, hands would occlude the camera,
+actions would be happening off-screen, and its hard to focus on one place if
+the camera is always moving away. The Spectator Mode lets you free move around
+the scene and view at whatever angle or focus on whatever area.
+
+![Spectator Mode](https://cloud.githubusercontent.com/assets/674727/24733331/f2bc6094-1a2b-11e7-9442-bcf30d18af79.gif)
```

Comment:
Fine to omit webgl insight, actually I only use that for looking at buffers and shaders, which is way out of scope here. Proposed: 

**three.js Inspector**

To understand the relationship between A-Frame components and the three.js scene graph, the [three.js inspector](https://chrome.google.com/webstore/detail/threejs-inspector/dnhjfclbfhcbcdfpjaeacomhbdfjbebi?hl=en) Chrome extension can be helpful. It shows the complete list of objects in your scene, and lets you modify THREE.Material properties directly temporarily.

![screen shot 2017-05-04 at 8 44 49 pm](https://cloud.githubusercontent.com/assets/1848368/25733151/28aa493c-310b-11e7-9400-6c3391cd7b12.png)
> Screenshot of three.js inspector.